pulsar-broker-common/src/main/java/org/apache/pulsar/broker/authentication/metrics/AuthenticationMetrics.java

```diff
@@ -32,6 +32,36 @@
             .labelNames("provider_name", "auth_method", "reason")
             .register();
 
+    private static final Counter clientCertSelfSignedMetrics = Counter.build()
+            .name("pulsar_authentication_tls_cert_self_signed_count")
+            .help("Pulsar client authenticating with a TLS cert that is self-signed")
+            .register();
+
+    private static final Counter clientCertSmallRsaKeySizeMetrics = Counter.build()
```

Comment:
can we make it generic instead specific to only rsa? `clientCertInvalidKeySizeMetrics`?

Reply:
Yes, I changed the metric to clientCertInvalidKeySizeMetrics. However, I added an 'algorithm' label to the metric to differentiate between 'rsa' and other types of key pairs since suggested key sizes differ greatly between RSA and ECC.

FYI based on research for [Issue 11431](https://github.com/apache/pulsar/issues/11431) it looks like the code for loading certs sort of forces users into using RSA key pairs. But it may be possible to change that in the future, also may be possible to load ECC key pair using an input stream (not sure)